src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,171 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<T>(null, value, null);
+        if (first == null) {
+            first = newNode;
+            last = newNode;
```

Comment:
you have the same line below, let's put it after if else and remove duplicate from else